doozer/doozerlib/distgit.py

```diff
@@ -1394,10 +1394,66 @@ def _clean_repos(self, dfp):
                 if changed:
                     dfp.add_lines_at(entry, "RUN " + new_value, replace=True)
 
-    def _mapped_image_from_member(self, image, original_parents, count):
+    def _resolve_parent(self, original_parent, dfp):
+        try:
+            self.logger.debug('Retrieving image info for image %s', original_parent)
+            cmd = f'oc image info {original_parent} -o json'
+            out, _ = exectools.cmd_assert(cmd, retries=3)
+            labels = json.loads(out)['config']['config']['Labels']
+
+            # Get the exact build NVR
+            build_nvr = f'{labels["com.redhat.component"]}-{labels["version"]}-{labels["release"]}'
+
+            # Query Brew for build info
+            self.logger.debug('Retrieving info for Brew build %s', build_nvr)
+            with self.runtime.shared_koji_client_session() as koji_api:
+                if not koji_api.logged_in:
+                    koji_api.gssapi_login()
+                build = koji_api.getBuild(build_nvr, strict=True)
+
+            # Get the pullspec for the upstream equivalent
+            upstream_equivalent_pullspec = build['extra']['image']['index']['pull'][1]
+
+            # Verify whether the image exists
+            self.logger.debug('Checking for upstream equivalent existence, pullspec: %s', upstream_equivalent_pullspec)
+            cmd = f'oc image info {upstream_equivalent_pullspec} --filter-by-os linux/amd64 -o json'
+            out, _ = exectools.cmd_assert(cmd, retries=3)
+
+            # It does. Use this to rebase FROM directive
+            mapped_image_data = json.loads(out)
+            mapped_image_data_labels = mapped_image_data['config']['config']['Labels']
+            mapped_image = f'{labels["name"]}:{mapped_image_data_labels["version"]}-{mapped_image_data_labels["release"]}'
```

Comment:
This will always be the same as the values in line 1405?
`build_nvr = f'{labels["com.redhat.component"]}-{labels["version"]}-{labels["release"]}'`
so we can just use those

Reply:
Labels slightly differ because `original_parent` is an image we mirror. Nonetheless, `release` and `version` appear to be the same:

```
10,11c10,11
<     "io.k8s.description": "ART equivalent image openshift-4.15-golang - rhel-8/golang",
<     "io.k8s.display-name": "builder-rhel-8-golang-1.20-openshift-4.15",
---
>     "io.k8s.description": "golang builder image for Red Hat internal builds",
>     "io.k8s.display-name": "Go Builder 1.20",
14,15d13
<     "io.openshift.build.name": "builder-rhel-8-golang-1.20-openshift-4.15--art-builder-474",
<     "io.openshift.build.namespace": "ci",
```

so I think it's fine to reuse the labels. `mapped_image` will need `name` instead of `com.redhat.component` though. 

Done in a809cf2bebdd0406c6d793da33902125afcaecd2